ostd/src/irq/level.rs

```diff
@@ -41,23 +60,48 @@ impl InterruptLevel {
         let level = INTERRUPT_LEVEL.load();
         match level {
             0 => Self::L0,
-            1 => Self::L1,
+            1 => {
+                let cpu_priv_at_irq = {
+                    let raw_cpu_priv = CPU_PRIV_AT_IRQ.load();
```

Comment:
This requires an additional CPU-local load in `InterruptLevel::current`, even though the caller does not care about the privilege information. While performance may not become the bottleneck, this approach is not ideal from an API design standpoint.

Why don't you like the previously proposed approach that squashes the privilege information into `INTERRUPT_LEVEL: CpuLocal<u8>`? It is more elegant because it avoids unnecessary CPU-local loads and stores in `call_irq_callback_functions`.

Another alternative is to keep the current approach, but add a standalone `InterruptLevel::is_kernel_interrupted` method that loads from `CPU_PRIV_AT_IRQ`. However, we'll still have more CPU-local loads and stores in `call_irq_callback_functions`.